hnn_core/network_builder.py

```diff
@@ -415,169 +415,50 @@ def _create_cells_and_feeds(self, threshold, record_vsoma=False,
-                                nc_dict)
-
-        # loop over _all_ drives, _connect_celltypes picks ones on this rank
-        for drive in self.net.external_drives.values():
+        assert len(self.cells) == len(self._gid_list) - len(self._drive_cells)
+        # Gather indeces of targets on current node
+        connectivity_list = [conn for conn in connectivity_list if
+                             _PC.gid_exists(conn['target_gid'])]
+        target_gids = [conn['target_gid'] for conn in connectivity_list]
+        target_filter = {}
+        for idx in range(len(self.cells)):
+            gid = self._gid_list[idx]
+            if gid in target_gids:
+                target_filter[gid] = idx
+
+        for conn in connectivity_list:
+            src_type, src_gid = conn['src_type'], conn['src_gid']
+            target_type, target_gid = conn['target_type'], conn['target_gid']
+            loc, receptor = conn['loc'], conn['receptor']
+            nc_dict = conn['nc_dict']
+            target_cell = self.cells[target_filter[target_gid]]
+            connection_name = f'{_short_name(src_type)}_'\
+                              f'{_short_name(target_type)}_{receptor}'
+            if connection_name not in self.ncs:
+                self.ncs[connection_name] = list()
+            pos_idx = src_gid - net.gid_ranges[_long_name(src_type)][0]
+            # NB pos_dict for this drive must include ALL cell types!
+            nc_dict['pos_src'] = net.pos_dict[
+                _long_name(src_type)][pos_idx]
+
+            # get synapse locations
+            syn_keys = list()
+            if loc in ['proximal', 'distal']:
+                for sect in target_cell.sect_loc[loc]:
+                    syn_keys.append(f'{sect}_{receptor}')
+            else:
+                syn_keys = [f'{loc}_{receptor}']
 
-            receptors = ['ampa', 'nmda']
-            if drive['type'] == 'gaussian':
-                receptors = ['ampa']
-            # conn-parameters are for each target cell type
-            for target_cell_type, drive_conn in drive['conn'].items():
-                for receptor in receptors:
-                    if len(drive_conn[receptor]) > 0:
-                        nc_dict['lamtha'] = drive_conn[
-                            receptor]['lamtha']
-                        nc_dict['A_delay'] = drive_conn[
-                            receptor]['A_delay']
-                        nc_dict['A_weight'] = drive_conn[
-                            receptor]['A_weight']
-                        self._connect_celltypes(
-                            drive['name'], target_cell_type,
-                            drive_conn['location'], receptor, nc_dict,
-                            unique=drive['cell_specific'])
+            for syn_key in syn_keys:
+                nc = target_cell.parconnect_from_src(
+                    src_gid, nc_dict, target_cell.synapses[syn_key])
+                self.ncs[connection_name].append(nc)
```

Comment:
how difficult would it be to add a test that checks that `self.ncs` has been updated when you do `add_connection` ? Ideally without running the simulation ...

Reply:
There's a `self.ncs` check for the drives API. Since the entire network is built I'll add a check for a subset of the connections.

Reply:
On a second look it seems like this is covered by: 
https://github.com/ntolley/hnn-core/blob/ea6e4407c4b672869509d446c4a129b00fef3e9b/hnn_core/tests/test_network.py#L144

The most recent update has `_set_default_connections()` call `add_connection()` so I don't think it's necessary to explicitly call it in `test_network.py`. Pretty easy to add the test if you disagree so just let me know.

Reply:
explicit test might still be helpful ... add one connection and check if that exists